### ISSUE TITLE
[Feature Request] Support spell points for class option per DMG pp. 288-89

### ISSUE THREAD

**Issue by caewok** [Orcpub/orcpub#440]
**Motivation / Use Case**

DMG pages 288 to 289 suggest an alternative set of casting rules based on spell points. This would be a useful addition because it could replace the casting rules for any magic using class and open up interesting options for different types of spellcasters. Right now, I am using points in my games but it would be nice to have actual character sheet support for it. 

**Expected Behavior**

1. Allow the user to define a certain number of points to cast a spell of a given level. Include 0 points as an option (e.g., for cantrips). Default to the DMG, but allow modifications.
2. Allow the user to define the total number of points for the class at each level, preferably including the choice of an attribute bonus if applicable.  Default to the DMG, but allow modifications.
3. Allow the user to track points used; reset points for short or long rest.
4. Allow the user to track number of uses of higher level spells, per the DMG suggestion to limit the number of higher level spells cast.

**Comment by DivertedCircle** [comment]
I believe adding the Spell Points system would violate the SRD since it is in the DMG only.

**Comment by caewok** [comment]
Fair point. Perhaps there is a way to do this that would not require the DMG, but instead just open up more options for homebrews? It really boils down to (1) use points instead of spell slots and (2) track points used.

**Comment by codeGlaze** [comment]
The ability to Homebrew something like this is a solid idea.

It has clear crossover merit with other ways to create point-like class features.

Difficulty of implementation is the bug question. 🤔

**Comment by caewok** [comment]
A couple of observations on what would need to change:
- In the character builder, the Spells selection does not change. The class can define, as before, number of cantrips and spells chosen per level using existing methods (e.g., druid vs. sorcerer vs. wizard spell selection methods). 
- In the character view, the Spell section should display near the top the number of points required at each level. It would look very similar to how the Slots view shows spell slots per level. If you had multiple classes using points, presumably each would be displayed separately, as you have no way of knowing how to mix them. The number of points for each spell level would be defined by the class, similar to how a class can define number of spell slots. 
- In the character view, the Spell section should display the total number of points for the class at the given level, comparable to how the Spell Preparation table displays it now. Consider using the Hit Points display, where you have a current number that can be modified and a maximum number that cannot. Total number of points would be defined by the class.  
- In the character view, casting a spell attributed to the points class source should subtract (or offer to subtract) points from the current points total for that class. 

I would think a similar set-up could be used for points-based features, in the Features tab of the character view. Instead of class source as with spells, you could have a table with feature source and an associated current number of points and maximum points for each feature. Not sure if that would be better or worse than the current checkbox approach for feature points. 

One bonus to this would be the ability to define a conversion between points and spell slots, as that is what the sorcerer class does. Automating that conversion would be helpful to sorcerers, instead of just listing the flexible casting as a descriptive feature.
